### ISSUE TITLE
[tabs] a11y review

### ISSUE THREAD

**Issue by Westbrook** [adobe/spectrum-web-components#1280]
### Feedback

- This is only half the tabs pattern. The tabpanel needs to be associated with the active tab. “Authors SHOULD associate a tabpanel element with its tab, either by using the aria-controls attribute on the tab to reference the tab panel, or by using the aria-labelledby attribute on the tab panel to reference the tab.” I would recommend the 2nd choice.

### Path forward

1. Document current usage state
2. Update to include #676 and #1039

cc @jnurthen

**Comment by Westbrook** [comment]
Looking into the API that would solve this issue as well as #676 and #1039 and have the following to propose:

cc: @davidclark87 @adixon-adobe @najikahalsema @ckundo @jnurthen 

## Proposal

```
<sp-tabs selected="1" auto>
    <sp-tab label="Tab 1" value="1"></sp-tab>
    <sp-tab label="Tab 2" value="2"></sp-tab>
    <sp-tab value="3">Tab 3</sp-tab>
    <sp-tab value="4">Tab 4</sp-tab>
    <sp-tab-panel value="1">Content for Tab 1</sp-tab-panel>
    <sp-tab-panel value="2">Content for Tab 2</sp-tab-panel>
    <sp-tab-panel value="3">Content for Tab 3</sp-tab-panel>
    <sp-tab-panel value="4">Content for Tab 4</sp-tab-panel>
</sp-tabs>
```

Here we would surface the ability to manage items in the `role=tablist` and items that would receive `role=tabpanel` with a minimized API surface. This would supply direct support for the second half of the tabs pattern listed as missing above while serving to provide clear usage direction to manage the additional content. This doesn't directly offer the ability to slot a child into a "tab" as initially asked in #676 or as offered in related Spectrum libraries, but it does reduce the required attribute API for one matching inclusion of the `value` attribute, which feels like enough of a trade off to make it worth it. This also allows a user to omit the `aria` and `role` attributes while still receiving them in the final rendered HTML.

## Docs
More complete documentation for the usage of this can be found at:
https://westbrook-tab-panel--spectrum-web-components.netlify.app/components/tabs
https://westbrook-tab-panel--spectrum-web-components.netlify.app/components/tab
https://westbrook-tab-panel--spectrum-web-components.netlify.app/components/tab-panel

## Question 1
Any issues with leveraging the `value` attribute, which otherwise informs the `sp-tabs` element what to set as `selected` as the tie that binds these two elements together?

## Question 2
Reviewing this with a screen reader pointed to one major difference between this output and what's found at https://w3c.github.io/aria-practices/examples/tabs/tabs-1/tabs.html the screen reader doesn't stop on the `role="tablist"` element in this implementation, is that wrong?

It does seem that other Spectrum implementations have this same reality, so it may signal something larger about the way the pattern is consumed/exposed by Chrome or macOS Voice Over?

## Question 3 
As per #1039 the ability to automatically activate a tab on "focus" has been added via the `auto` attribute. This is currently activated via the `Arrow*` key event, as found in the w3c link above, but it doesn't occur when using `control + option + Arrow*` via the screen reader to go between the available tabs. The w3c sample, as well as other Spectrum implementations, seem to exclude this interaction as well. Should `auto` expand to this interaction: e.g. `sp-tab` receiving focus?

## More questions?
Please let me know if you can think of any other contexts that we should attempt to cover/avoid with this sort of an update to the element API of this pattern.

**Comment by ckundo** [comment]
> This is currently activated via the Arrow* key event, as found in the w3c link above, but it doesn't occur when using control + option + Arrow* via the screen reader to go between the available tabs

this may depend on VoiceOver configuration. Make sure that browser focus is following the VoiceOver cursor (you can check in VoiceOver Utility under Web).

**Comment by ckundo** [comment]
@Westbrook I think I understand your question now. VO + arrow will move focus, and with `auto` enabled, moving focus should also move selection. So instead of listening to arrow keys to change the selection (which seems like the w3c example does), in my opinion you should listen for focus change event and move the selection based on that.

**Comment by jnurthen** [comment]
> Question 2
> Reviewing this with a screen reader pointed to one major difference between this output and what's found at https://w3c.github.io/aria-practices/examples/tabs/tabs-1/tabs.html the screen reader doesn't stop on the role="tablist" element in this implementation, is that wrong?

I suspect that if you add an `aria-label` on the tablist a screen reader would find something interesting to speak and vocalize it. This seems to be the major implementation difference

**Comment by jnurthen** [comment]
I disagree with @ckundo about the focus. I do NOT want selected tab to change in the case when I move VO focus. If this were the case then I would be unable to interact with anything but the last tab page as focus would change as I simply browse through the page. The behaviour in the APG pattern is correct.

**Comment by Westbrook** [comment]
@jnurthen `aria-label` does trigger that. Thanks! Would you count that are a "required" attribute for the `role="tablist"` element?

It _was_ with the `aria-label` attribute that I was discussing this interaction with @ckundo in that `shift + control + option + ArrowRight` will stop on the list, but only traverse to the individual tab with a `shift + control + option + ArrowDown` or `control + option + ArrowRight` interaction. After selecting a tab with focus the `shift + control + option + ArrowUp` interaction brings you back to the list without changing the panel selection. A lot of both nuance and option as to how to navigate to this content, so tricky to fully know the depth to which something like https://w3c.github.io/aria-practices/#kbd_selection_follows_focus applied to the various forms of "focus" in the interaction.

**Comment by jnurthen** [comment]
https://w3c.github.io/aria-practices/#naming_role_guidance has handy guidance on naming by role. It states "recommended" 

For navigation remember that it is not just VO we are talking about. Other screen readers do not have the VO grouping concept and will only allow navigation in a linear manner. IMO it is unexpected if selection changes when merely navigating the page.

**Comment by Westbrook** [comment]
Ah, cool. That totally makes sense if the grouping concept if VO-centric. I've gotta bone up on the alternative screen readers!

Two questions down, one to go. Thanks to both of you!

**Comment by jnurthen** [comment]
For Q1 I don't think `value` is a problem
